semeio/jobs/scripts/fm_pyscal.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python
+"""Forward model for connecting ERT with the Pyscal command line client"""
+import logging
+import sys
+import os
+import argparse
+
+import six
+
+from semeio.jobs.design_kw.design_kw import extract_key_value
+
+from pyscal import pyscalcli
+
+_logger = logging.getLogger("FM_PYSCAL")
+
+# The string used here must match what is used as the DEFAULT
+# parameter in semeio/jobs/config_jobs/PYSCAL. It is not used elsewhere.
+MAGIC_NONE = "__NONE__"
+
+# These key-values are added to the dictionary parsed from
+# parameters.txt, and allow supplying magic interpolation values
+# directly from the ert config file.
+MAGIC_CASES = {
+    "__BASE__": 0,
+    "__LOW__": -1,
+    "__PESS__": -1,
+    "__PESSIMISTIC__": -1,
+    "__HIGH__": 1,
+    "__OPT__": 1,
+    "__OPTIMISTIC__": 1,
+}
+
+
+def main_entry_point(args=None):
+    """This mimics the pyscal command line client, but differs because
+    all arguments are required (but can be defaulted using MAGIC_NONE)"""
+    parser = _get_args_parser()
+    options = parser.parse_args(args)
+    run(
+        options.relperm_parameters_file,
+        options.output_filename,
+        options.sheet_name,
+        options.int_param_wo_name,
+        options.int_param_go_name,
+        options.slgof,
+        options.family,
+    )
+
+
+def _get_args_parser():
+    """Construct an argparse parser for fm_pyscal"""
+    description = (
+        "ERT forward model argument parser for wrapper around pyscal. "
+        "This parser should only be used in the forward model context, "
+        "where all arguments must always be passed. For other uses, "
+        "head to the pyscal client from the pyscal package"
+    )
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("relperm_parameters_file", type=str)
+    parser.add_argument("output_filename", type=str)
+    parser.add_argument("sheet_name", type=str)
+    parser.add_argument("int_param_wo_name", type=str)
+    parser.add_argument("int_param_go_name", type=str)
+    parser.add_argument("slgof", type=str)
+    parser.add_argument("family", type=int)
+    return parser
+
+
+def run(
```

Comment:
Can we split this function into multiple private functions? It is a tad long ;)

Reply:
Refactored out one function.